### DISCUSSION TITLE
FastAPI with many-to-many association

### DISCUSSION THREAD

**Discussion by rikbevelsborg** [fastapi/fastapi#2718] [Questions]
Hi all,

Thanks in advance for FastAPI, what a beautiful package! 

I am running into an trouble when I try to do CRUD operations with SQLAlchemy when I combine two models with an association table / association class as a many-to-many. How do you have to define a create operation, use schemas and set up a post route for this case? 

Hope you could help me out here!

`class User(Base):
    __tablename__ = 'user'

    id = sa.Column(sa.Integer, primary_key=True, index=True)
    email = sa.Column(sa.String, unique=True, index=True, nullable=False)
    first_name = sa.Column(sa.String)
    last_name = sa.Column(sa.String)
    hashed_password = sa.Column(sa.String, nullable=False)
    is_active = sa.Column(sa.Boolean, default=True)
    is_superuser = sa.Column(sa.Boolean, default=False)

    # Relationships
    profile = relationship("Profile", uselist=False, back_populates="user_profile", cascade="all, delete-orphan") # One-to-One with Profile
    

class Profile(Base):
    __tablename__ = "profile"

    id = sa.Column(sa.Integer, primary_key=True, index=True)
    user_id = sa.Column(sa.Integer, sa.ForeignKey("user.id"))
    
    # Relationships
    buckets = relationship("Bucket", secondary="profile_bucket") # Many-to-Many 

class Bucket(Base):
    __tablename__ = "bucket"

    id = sa.Column(sa.Integer, primary_key=True, index=True)
    bucket_name = sa.Column(sa.String(100))

    # Relationships
    profile = relationship("Profile", secondary="profile_bucket", viewonly=True) # Many-to-Many

class ProfileBucket(Base):
    __tablename__ = 'profile_bucket'

    profile_id = sa.Column('profile_id', sa.ForeignKey('profile.id'), primary_key=True)
    bucket_id = sa.Column('bucket_id', sa.ForeignKey('bucket.id'), primary_key=True)

    profile = relationship("Profile", backref=backref("profile_bucket", cascade="all, delete-orphan" ))
    bucket = relationship("Bucket", backref=backref("profile_bucket", cascade="all, delete-orphan" ))`

**Comment by Mause** [comment]
Have you followed the tutorials in the documentation at all? This shouldn't be difficult to extrapolate

**Reply by rikbevelsborg** [reply]
Solved it
